journalpump/journalpump.py

```diff
@@ -495,6 +509,37 @@ def value_func(tags, f=f, args=args):  # pylint: disable=undefined-variable
 
             yield output
 
+    def _configure_secret_filter_metrics(self, config):
+        secret_filter_metrics = config.get("secret_filter_metrics")
+        if secret_filter_metrics is not None and secret_filter_metrics is True:
+            return True
+        return False
+
+    def _validate_and_build_secret_filters(self, config):
+        secret_filters = config.get("secret_filters")
+        if secret_filters is None:
+            return []
+
+        if not isinstance(secret_filters, list):
+            raise ValueError("invalid secret_filters configuration - must be a list")
+
+        for secret_filter in secret_filters:
+
+            if secret_filter.get("redaction_message") is None:
+                raise ValueError("invalid secret_filters configuration - missing field 'redaction_message'")
+
+            if secret_filter.get("pattern") is None:
+                raise ValueError("invalid secret_filters configuration - missing field 'pattern'")
+
+        # Compile / validate regex
+        for idx, _ in enumerate(secret_filters):
+            try:
+                secret_filters[idx]["compiled_pattern"] = re.compile(secret_filters[idx]["pattern"])
+            except:
+                raise ValueError("invalid secret_filters configuration - invalid regex") from None
```

Comment:
Is there some particular reason to hide the original error here? Could contain useful information.

Reply:
fixed